src/integritysetup/integrity-util.c

```diff
@@ -34,9 +34,18 @@ int parse_integrity_options(
                 else if (streq(word, "allow-discards")) {
                         if (ret_activate_flags)
                                 *ret_activate_flags |= CRYPT_ACTIVATE_ALLOW_DISCARDS;
-                } else if (streq(word, "no-journal")) {
-                        if (ret_activate_flags)
-                                *ret_activate_flags |= CRYPT_ACTIVATE_NO_JOURNAL;
+                } else if ((val = startswith(word, "mode="))) {
```

Comment:
might be good to do something sensible when `mode=` is passed multiple times - either reject outright with an error, or clear the flags and set them again so that the latest one wins

Reply:
we usually follow the logic that the last setting wins (there are exceptions though)

hence, basically, if you set CRYPT_ACTIVATE_NO_JOURNAL_BITMAP then turn off CRYPT_ACTIVATE_NO_JOURNAL and so on.

Reply:
Sure, I can implement last one wins. Which fixup procedure do you favour, a fixup commit or a force-push?

Reply:
we prefer "perfect" PRs, i.e. where each commit makes logical sense and stands on its own feet and makes sense on its own. It should not necessarily reflect historical changes thus.

or in other words we want full "bisectability", and we don't want that later commits of one PR undo changes from earlier commits of the same PR.